h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ChunkConverter.java

```diff
@@ -69,6 +71,106 @@ long getCurrentRecordIdx() {
     return _currentRecordIdx;
   }
 
+  private PrimitiveConverter nullConverter(byte vecType, PrimitiveType parquetType) {
+    switch (vecType) {
+      case Vec.T_BAD:
+      case Vec.T_CAT:
+      case Vec.T_STR:
+      case Vec.T_UUID:
+      case Vec.T_TIME:
+        if (OriginalType.TIMESTAMP_MILLIS.equals(parquetType.getOriginalType()) || parquetType.getPrimitiveTypeName().equals(PrimitiveType.PrimitiveTypeName.INT96)) {
+          return null;
+        } else {
+          boolean dictSupport = parquetType.getOriginalType() == OriginalType.UTF8 || parquetType.getOriginalType() == OriginalType.ENUM;
+          return new NullStringConverter(dictSupport);
+        }
+      case Vec.T_NUM:
+        if (OriginalType.DECIMAL.equals(parquetType.getOriginalType()))
+          return new NullDecimalConverter();
+        else
+          return new NullNumberConverter();
+      default:
+        throw new UnsupportedOperationException("Unsupported type " + vecType);
+    }
+  }
+
+  private static class NullNumberConverter extends PrimitiveConverter {
+
+    @Override
+    public void addBoolean(boolean value) { }
+
+    @Override
+    public void addDouble(double value) { }
+
+    @Override
+    public void addFloat(float value) { }
+
+    @Override
+    public void addInt(int value) { }
+
+    @Override
+    public void addLong(long value) { }
+
+    @Override
+    public void addBinary(Binary value) { }
+  }
+
+  private static class NullDecimalConverter extends PrimitiveConverter {
+
+    @Override
+    public void addBoolean(boolean value) {
+      throw new UnsupportedOperationException("Boolean type is not supported by DecimalConverter");
+    }
+
+    @Override
+    public void addDouble(double value) {
+      throw new UnsupportedOperationException("Double type is not supported by DecimalConverter");
+    }
+
+    @Override
+    public void addFloat(float value) {
+      throw new UnsupportedOperationException("Float type is not supported by DecimalConverter");
+    }
+
+    @Override
+    public void addInt(int value) { }
+
+    @Override
+    public void addLong(long value) { }
+
+    @Override
+    public void addBinary(Binary value) {
+      throw new UnsupportedOperationException("Arbitrary precision Decimal type is currently not supported by H2O." +
+              "Please use 64-bit decimal type instead (precision <= 18).");
+    }
+  }
+
+
+  private static class NullStringConverter extends PrimitiveConverter {
+    private final boolean _dictionarySupport;
```

Comment:
please move all of these to NullConverter

Reply:
Done